src/api/erc20/Erc20ApiProxy.ts

```diff
@@ -0,0 +1,58 @@
+import BN from 'bn.js'
+
+import { Receipt } from 'types'
+
+import {
+  Erc20Api,
+  NameParams,
+  SymbolParams,
+  DecimalsParams,
+  TotalSupplyParams,
+  BalanceOfParams,
+  AllowanceParams,
+  ApproveParams,
+  TransferParams,
+  TransferFromParams,
+} from './Erc20Api'
+import { CacheProxy } from 'api/proxy'
+
+const DEFAULT_CACHE_TIME = 30
```

Comment:
Why 30? at least the symbol, and so on, we can catch forever, right?

Reply:
I didn't get to test it, but one use case you mentioned before was that a token might be updated with those details?

Now that I'm thinking about it makes little sense, we could just cache those forever indeed.

Reply:
Done